### ISSUE TITLE
Document Snap-deals feature.

### ISSUE THREAD

**Issue by jennijuju** [filecoin-project/lotus-docs#48]
inputs we got from @ZenGround0, that can be used as our guide here: @jennijuju @johnnymatthews 

- [x] Explain the Main workflow: lotus-miner snap-up <sector-id> then wait for a deal, then FSM automatically takes it through replica update back to proving
- [x] snap-up gotchas: sector must be proven by wdpost first, sector must be in proving state in fsm, can't be too many sectors waiting for deals per config MaxWaitDealsSectors
- [x] Replica update gotchas: if you are unlucky and the fsm finishes the replica update in the deadline of the sector or the one before it will hang for a while in the ReplicaUpdateFailed state.  The fsm will keep retrying until the immutable deadlines pass and submit the message.  No action required in this case
- [x] New config option MakeNewSectorForDeals(in followup PR) makes it so that only snap deals will be accepted.  Deals will hang until you make a snap deal instead of kicking off a new sector
- [x] lotus-miner sectors mark-for-upgrade deprecated in v15
- [x] Extension workflow: if you have a snap deals sector waiting for deals and make a deal that is staged but that does not fit in that sector because the sector expires too soon the thing to do is ./lotus-miner sectors extend --new-expiration=<new-exp> <sector-id> and then ./lotus-miner sectors match-pending-pieces this will match the piece to the newly extended sector and start the replica update process
- [x] If the upgrade is going bad and you don't know how to proceed, say for example you are stuck in ReplicaUpdateFailed for more than two deadlines and recovery does not look like it will happen on its own you can abort the upgrade and remove all replica update data reverting your sector back to the proving state with ./lotus-miner sectors update-state --really-do-it <sector-id> AbortUpgrade .  One gotcha here: you will lose the deals in the update (the fsm won't rematch the deal with other sectors) the same way you lose them when removing a deal sector

**Comment by BlocksOnAChain** [comment]
@johnnymatthews Added the description above, just FYI.

**Comment by johnnymatthews** [comment]
Depends on #72.

**Comment by johnnymatthews** [comment]
Some stuff we need to explain:

- [ ] What Snap-deals is.
    - [ ] What problem are they trying to solve?
    - [ ] Which usergroup is mostly affected by this feature?
- [ ] Tutorial series:
    - [ ] Spin up a dev-net.
    - [ ] Create some basic deals.
    - [ ] Convert those deals to snap-deals.
    - [ ] Final video of one of the devs doing this on **mainnet**.

**Comment by BlocksOnAChain** [comment]
issue outcomes are on this page: https://deploy-preview-76--blissful-liskov-5fa2cf.netlify.app/docs/storage-providers/snap-deals/
We are currently "in review" stage for this side of docs work.
@johnnymatthews FYI
